### ISSUE TITLE
Attempted to upgrade to latest bitcoinj and xchange, dies because no commons-logging

### ISSUE THREAD

**Issue by kenrestivo** [gary-rowe/multibit-merchant#79]
I was going to try to get MultiBitMerchant running (develop branch), and maven complained that there was no xchange 1.0.0  in Maven. Sure enough, that's ancient, and there isn't. While looking in the POM file I saw that it also used an ancient bitcoinj. So I updated the pom to use the latest xchange, and latest bitcoinj. And then it seems like someone refactored xchange and moved MoneyUtils around. So I fixed those.

Then the DAO tests were all failing, complaining about Spring. I don't know spring so I just whacked the tests in an attempt to get it to build. I did manage to get it to build!

Then I tried to run it. And I finally decided to cut my losses once I got this error.

```
$ java -jar target/mbm-develop-SNAPSHOT.jar server mbm.yml
INFO  [2013-03-04 08:36:38,057] org.multibit.mbm.MultiBitMerchantService: Reading configuration
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/commons/logging/LogFactory
    at org.springframework.context.support.AbstractApplicationContext.<init>(AbstractApplicationContext.java:161)
    at org.springframework.context.support.AbstractRefreshableApplicationContext.<init>(AbstractRefreshableApplicationContext.java:90)
    at org.springframework.context.support.AbstractRefreshableConfigApplicationContext.<init>(AbstractRefreshableConfigApplicationContext.java:59)
    at org.springframework.context.support.AbstractXmlApplicationContext.<init>(AbstractXmlApplicationContext.java:61)
    at org.springframework.context.support.ClassPathXmlApplicationContext.<init>(ClassPathXmlApplicationContext.java:136)
    at org.springframework.context.support.ClassPathXmlApplicationContext.<init>(ClassPathXmlApplicationContext.java:93)
    at org.multibit.mbm.MultiBitMerchantService.initialize(MultiBitMerchantService.java:65)
    at org.multibit.mbm.MultiBitMerchantService.initialize(MultiBitMerchantService.java:36)
    at com.yammer.dropwizard.AbstractService.initializeWithBundles(AbstractService.java:181)
    at com.yammer.dropwizard.cli.ServerCommand.run(ServerCommand.java:49)
    at com.yammer.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:98)
    at com.yammer.dropwizard.cli.Command.run(Command.java:111)
    at com.yammer.dropwizard.AbstractService.run(AbstractService.java:197)
    at org.multibit.mbm.MultiBitMerchantService.main(MultiBitMerchantService.java:48)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.logging.LogFactory
    at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:423)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:356)
    ... 14 more
```

I added commons-logging to the POM, but still got the same error.

The work I did is available here, in case anyone wants it, or to try to fix the commons-logging problem:
https://github.com/kenrestivo/MultiBitMerchant/commits/upgrade-to-modern

TL;DR: I'd love to have been able to have tested out MultiBitMerchant, but burned a couple hours trying to get it to build and then gave up.

**Comment by gary-rowe** [comment]
Thanks for trying it out. I'm going to be putting some more effort to it once I can free up time from other Bitcoin projects.

Were you working on the master branch or develop? I think the current master is broken with this issue (I'll resolve it tonight) but the develop branch should be good to go right out of the box.

**Comment by kenrestivo** [comment]
I was on develop. One can verify this by looking at my commit history; the commit from which I branched off was 7aec1e3, which is the HEAD of develop AFAICT.

**Comment by gary-rowe** [comment]
I've upgraded XChange and Bitcoinj and made the necessary dependency changes. You will need to build and install Bitcoinj directly from git (instructions are in the README.md)

Can you verify that this has fixed your issue, and if you're happy close the issue?

**Comment by kenrestivo** [comment]
Confirmed! It builds and runs! I upgraded to bitcoinj 0.7.1 too, will submit pull request. Thanks!
